<?xml version="1.0" encoding="utf-8"?>
<!--
Launch file for Selective ICP with rviz
-->

<launch>

  <!-- Proper Output Info -->
  <env name="ROSCONSOLE_FORMAT" value="[${severity}] (${node}) [${time}]: ${message}\n" />

  <param name="/use_sim_time" value="true" />
    <!--<arg name="cad_file" default="$(find cpt_selective_icp)/resources/cla.json" />-->
    <arg name="rosbag" />
    <arg name="logprefix" />
    <arg name="segmentation_model" />
    <arg name="input_height" />
    <arg name="input_width" />
    <arg name="threshold" default="150" />
 
    <node pkg="background_foreground_segmentation"
          type="inference_node2.py"
          name="inference_node"
          output="screen">
        <param name="model_gdrive_id" value="$(arg segmentation_model)" />
        <param name="input_height" value="$(arg input_height)" />
        <param name="input_width" value="$(arg input_width)" />
        <param name="pseudo_rgb" value="false" />
        <rosparam>
          image_topics:
            - /rgb_front
            - /rgb_back
            - /rgb_left
            - /rgb_right
          segmentation_output_topics:
            - /rgb_front/segmented
            - /rgb_back/segmented
            - /rgb_left/segmented
            - /rgb_right/segmented
        </rosparam>
    </node>

    <node pkg="background_foreground_segmentation"
          type="depth_filtering_node"
          name="depth_filter"
          output="screen">
        <param name="density_threshold" value="$(arg threshold)" />
        <rosparam param="image_topics">
            - /depth_front
            - /depth_back
            - /depth_left
            - /depth_right
        </rosparam>
        <rosparam param="caminfo_topics">
            - /pickelhaube/cam0/camera_info
            - /pickelhaube/cam1/camera_info
            - /pickelhaube/cam2/camera_info
            - /pickelhaube/cam3/camera_info
        </rosparam>
        <rosparam param='mask_topics'>
            - /rgb_front/segmented
            - /rgb_back/segmented
            - /rgb_left/segmented
            - /rgb_right/segmented
        </rosparam>
    </node>

    <!---->
    <!--<node type="rviz" name="rviz" pkg="rviz" args="-d $(find background_foreground_segmentation)/config/segmented_2.rviz" />-->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find background_foreground_segmentation)/config/4cloud.rviz" />
    <!-- should be started last for handover to ICP-->
    <!--<include file="$(find background_foreground_segmentation)/launch/alphasense_tools/pickelhaube_from_bag.launch">-->
    <include file="$(find background_foreground_segmentation)/launch/isaac_tools/isaac_from_bag.launch">
        <arg name="rosbag" value="$(arg rosbag)" />
    </include>

</launch>
